libs/dyn/visit_map.go

```diff
@@ -25,7 +25,7 @@ func Foreach(fn MapFunc) MapFunc {
 				if err != nil {
 					return InvalidValue, err
 				}
-				m.Set(pk, nv) //nolint:errcheck
```

Comment:
Is it an equivalent change? Won't pk now lose the location information if it had any?

Reply:
no, it won't update existing location information.

Reply:
see also 5facf29d57d52f0ed16e36981404922ef125bb54